presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2187,10 +2231,15 @@ public ConnectorTableLayoutHandle getAlternativeLayoutHandle(ConnectorSession se
         checkArgument(hiveLayoutHandle.getBucketHandle().isPresent(), "Hive connector only provides alternative layout for bucketed table");
         HiveBucketHandle bucketHandle = hiveLayoutHandle.getBucketHandle().get();
         ImmutableList<HiveType> bucketTypes = bucketHandle.getColumns().stream().map(HiveColumnHandle::getHiveType).collect(toImmutableList());
+        Optional<List<HiveType>> hiveTypes = hivePartitioningHandle.getHiveTypes();
```

Comment:
This can only be `HIVE_COMPATIBLE`, see explanation `checkArgument(
                hivePartitioningHandle.getBucketFunctionType().equals(HIVE_COMPATIBLE) &&
                        hiveTypes.isPresent(),
                "partitionColumns in HiveTableLayoutHandle must use a hive compatible bucket function");`

Reply:
As per discussion offline we decided to keep an assertion, as this method shouldn't be used for temporary tables